src/results.cpp

```diff
@@ -610,23 +607,31 @@ Results Results::sort(SortDescriptor&& sort) const
 
 Results Results::filter(Query&& q) const
 {
+    if (m_descriptor_ordering.will_apply_limit())
+        throw UnimplementedOperationException("Filtering a Results with a limit is not yet implemented");
```

Comment:
We could probably support this by moving the limit into the query itself in core, or emulating ordered behaviour at this level. It is most likely a rare use case though.